Common/src/main/java/com/blamejared/crafttweaker/api/zencode/IScriptLoadSource.java

```diff
@@ -0,0 +1,16 @@
+package com.blamejared.crafttweaker.api.zencode;
+
+import com.blamejared.crafttweaker.api.CraftTweakerAPI;
+import net.minecraft.resources.ResourceLocation;
+
+@FunctionalInterface
+public interface IScriptLoadSource {
+    
+    static IScriptLoadSource find(final ResourceLocation id) {
+        
+        return CraftTweakerAPI.getRegistry().findLoadSource(id);
+    }
+    
+    ResourceLocation id();
```

Comment:
Having it be a record means that `new` is as exposed as the class, so people could create new load sources at will, which is something I'd like to prevent